### ISSUE TITLE
If the header contains `!` (exclamation mark) then the type and subject are treated as empty

### ISSUE THREAD

**Issue by almaceleste** [joshbolduc/vscode-commitlint#789]
If the header contains `!` (exclamation mark) in `type/scope` part (to indicate breaking changes), then the type and subject are treated as empty.
This behavior occurs even if I disable `subject-exclamation-mark` rule:
```
        "subject-exclamation-mark": [0, "always"],
```

**Comment by joshbolduc** [comment]
Hmm, I can't seem to reproduce. Does the same thing happen if you use commitlint directly? If not, what version of commitlint do you have installed?

You could also try enabling `commitlint.preferBundledLibraries` and seeing if that changes anything.

**Comment by almaceleste** [comment]
`commitlint` CLI finds no errors in such `COMMIT_MESSAGE` file.
but `commitliint` extension does.
enabled `commitlint.preferBundledLibraries` option does not change this behavior.

commitlint CLI: 19.6.0
commitlint extension: 2.6.0
vscode: 1.95.3

**Comment by joshbolduc** [comment]
I still don't see this behavior myself, but there is a [suspiciously similar sounding issue in the commitlint repo](https://github.com/conventional-changelog/commitlint/issues/4191).

**Comment by almaceleste** [comment]
yeah, looks similar, but I do not use `semantic-release` yet, and in my case it's the extension that complains, not the `commitlint` itself.
